toolchain/check/type_structure.cpp

```diff
@@ -132,7 +132,7 @@ class TypeStructureBuilder {
     BuildTypeStructure();
 
     return TypeStructure(std::exchange(structure_, {}),
-                         first_symbolic_distance_);
+                         std::exchange(symbolic_type_indices_, {}));
```

Comment:
This and the `TypeStructure` constructor will be performing quite a few `SmallVector` moves, which aren't cheap because `SmallVector` is not an especially small type.

Hm. The sort we perform in `impl_lookup` will also be doing a bunch of `SmallVector` moves too.

... OK, let's go with this for now, but can you add a TODO somewhere to reduce the number of `SmallVector` moves we perform?

Reply:
OK, done.